assets/tests/components/lateView.test.tsx

```diff
@@ -880,59 +774,35 @@ describe("LateView", () => {
       ],
     }
 
-    const firstLoadWrapper = mount(
+    const result = render(
       <StateDispatchProvider state={state} dispatch={jest.fn()}>
         <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
           <LateView />
         </VehiclesByRouteIdProvider>
       </StateDispatchProvider>
     )
 
-    act(() => {
-      firstLoadWrapper
-        .find(
-          '.m-late-view__missing-logons .m-late-view__data-row input[type="checkbox"]'
-        )
-        .first()
-        .simulate("click")
-
-      firstLoadWrapper
-        .find(
-          '.m-late-view__late-buses .m-late-view__data-row input[type="checkbox"]'
-        )
-        .first()
-        .simulate("click")
-
-      firstLoadWrapper
-        .find(".m-late-view__hide-popup button")
-        .first()
-        .simulate("click")
-    })
-    expect(
-      firstLoadWrapper.find(".m-late-view__data-row--selected")
-    ).toHaveLength(0)
+    await userEvent.click(
+      result.getByTestId(`row-checkbox-${missingLogonGhost1.runId}`)
+    )
+    await userEvent.click(
+      result.getByTestId(`row-checkbox-${lateVehicle1.runId}`)
+    )
+    await userEvent.click(result.getByRole("button", { name: /Hide/ }))
 
-    expect(
-      firstLoadWrapper.find(".m-late-view__data-row--unselected")
-    ).toHaveLength(2)
+    expect(result.getAllByTestId(/row-checkbox/)).toHaveLength(2)
 
-    const reloadWrapper = mount(
+    result.rerender(
       <StateDispatchProvider state={state} dispatch={jest.fn()}>
         <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
           <LateView />
         </VehiclesByRouteIdProvider>
       </StateDispatchProvider>
     )
-    expect(reloadWrapper.find(".m-late-view__data-row--selected")).toHaveLength(
-      0
-    )
-
-    expect(
-      reloadWrapper.find(".m-late-view__data-row--unselected")
-    ).toHaveLength(2)
+    expect(result.getAllByTestId(/row-checkbox/)).toHaveLength(2)
```

Comment:
If I'm reading things correctly, the checkboxes get the `row-checkbox` test ID regardless of whether or not they're selected, so I think the assertion may need to be refined a bit to match the original test.

Reply:
Great catch, I'll update! I did think about including a notion of whether or not the row is selected in the testid, but opted not to in favor of using the `toHaveProperty` test. This is an example of where that choice seemed to get a bit clunkier